In [1]:
#Declaracion de bibliotecas a utilizar
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [2]:
#Lectura del archivo de texto
df0 = pd.read_csv('bahiakino.txt', skiprows=0, sep='\s+') 
df0 = df0.replace('Nulo','NA')

In [3]:
#Conversion de los datos del archivo en un marco de datos
df=pd.DataFrame(df0)

In [4]:
#Tipos de datos de la tabla
df.dtypes

FECHA     object
PRECIP    object
EVAP      object
TMAX      object
TMIN      object
dtype: object

In [5]:
#Conversion de la columna "Fecha" en una variable
df['FECHA'] = pd.to_datetime(df.apply(lambda x: x['FECHA'],1), dayfirst=True)

In [6]:
#Conversion de las columnas  "PRECIP", "EVAP", "TMAX", "TMIN" en variables numericas
cols = ['PRECIP', 'EVAP', 'TMAX', 'TMIN']
df[cols] = df[cols].apply(pd.to_numeric, errors='coerce', axis=1)

In [7]:
df.dtypes

FECHA     datetime64[ns]
PRECIP           float64
EVAP             float64
TMAX             float64
TMIN             float64
dtype: object

In [8]:
#Inicio de tabla de datos
df.head()

FECHA  PRECIP  EVAP  TMAX  TMIN
0 1974-07-01     0.0   7.3  35.0  22.0
1 1974-07-02     0.0  10.5  34.0  22.5
2 1974-07-03     0.0  13.6  31.5  21.0
3 1974-07-04     0.0  10.7  41.0  28.0
4 1974-07-05     0.0  10.9  35.0  28.0

In [9]:
#Final de tabla de datos 
df.tail()

FECHA  PRECIP  EVAP  TMAX  TMIN
6630 2011-07-27     0.0   9.4  34.0  24.0
6631 2011-07-28     0.0   7.0  33.5  24.5
6632 2011-07-29     0.0   7.6  33.0  25.5
6633 2011-07-30     0.0   8.1  36.0  25.5
6634 2011-07-31     0.0   NaN   NaN  27.5

In [10]:
#Promedio
df.mean()

PRECIP     0.358136
EVAP       6.096420
TMAX      27.111920
TMIN      13.964030
dtype: float64

In [11]:
#Desviacion estandar
df.std()

PRECIP    2.788885
EVAP      2.714726
TMAX      5.785450
TMIN      7.048581
dtype: float64

In [12]:
#Mediana
df.median()

PRECIP     0.0
EVAP       6.0
TMAX      27.0
TMIN      12.5
dtype: float64

In [13]:
#Valor maximo de cada columna
df.max()

FECHA     2011-07-31 00:00:00
PRECIP                     70
EVAP                     17.7
TMAX                     48.5
TMIN                       30
dtype: object

In [14]:
#Valor minimo de cada columna
df.min()

FECHA     1974-07-01 00:00:00
PRECIP                      0
EVAP                      0.1
TMAX                     10.5
TMIN                       -4
dtype: object

In [15]:
df.describe()

PRECIP         EVAP        TMAX         TMIN
count  6619.000000  4721.000000  6485.00000  6486.000000
mean      0.358136     6.096420    27.11192    13.964030
std       2.788885     2.714726     5.78545     7.048581
min       0.000000     0.100000    10.50000    -4.000000
25%       0.000000     4.100000    22.00000     8.000000
50%       0.000000     6.000000    27.00000    12.500000
75%       0.000000     7.900000    32.00000    20.000000
max      70.000000    17.700000    48.50000    30.000000

In [16]:
#Creacion de variables de mes y año
df['MES'] = df['FECHA'].dt.month 
df['AÑO'] = df['FECHA'].dt.year

In [17]:
df.head(),df.tail()

(       FECHA  PRECIP  EVAP  TMAX  TMIN  MES   AÑO
 0 1974-07-01     0.0   7.3  35.0  22.0    7  1974
 1 1974-07-02     0.0  10.5  34.0  22.5    7  1974
 2 1974-07-03     0.0  13.6  31.5  21.0    7  1974
 3 1974-07-04     0.0  10.7  41.0  28.0    7  1974
 4 1974-07-05     0.0  10.9  35.0  28.0    7  1974,
           FECHA  PRECIP  EVAP  TMAX  TMIN  MES   AÑO
 6630 2011-07-27     0.0   9.4  34.0  24.0    7  2011
 6631 2011-07-28     0.0   7.0  33.5  24.5    7  2011
 6632 2011-07-29     0.0   7.6  33.0  25.5    7  2011
 6633 2011-07-30     0.0   8.1  36.0  25.5    7  2011
 6634 2011-07-31     0.0   NaN   NaN  27.5    7  2011)

In [18]:
# Número de años distintos
NumA = len(df['AÑO'].unique())
print(NumA, "años de datos")

24 años de datos


In [19]:
#Loop para contar el promedio de precipitaciones por mes
total=0.0
for i in range(12): #Contador de 1 a 12 para los meses
    PrecipMensual = df['PRECIP'][df.MES==[i+1]].sum()/NumA #SUmatoria de la precipitacion por mes 
                                                           #y divida entre el numero de años 
    total=total+PrecipMensual                              #Guardamos la cantidad calculada anteriormente
    print("Precipitacion del Mes", i+1,":", np.round(PrecipMensual, decimals=2), "mm", ", Precipitacion Acumulada:", 
          np.round(total, decimals=2), "mm") #Impresion del resultado del calculo de la precipitacion por mes
                                             #y precipitaciones acumuladas redondeadas a 2 decimales

Precipitacion del Mes 1 : 7.51 mm , Precipitacion Acumulada: 7.51 mm
Precipitacion del Mes 2 : 5.17 mm , Precipitacion Acumulada: 12.68 mm
Precipitacion del Mes 3 : 1.75 mm , Precipitacion Acumulada: 14.42 mm
Precipitacion del Mes 4 : 0.67 mm , Precipitacion Acumulada: 15.09 mm
Precipitacion del Mes 5 : 0.29 mm , Precipitacion Acumulada: 15.38 mm
Precipitacion del Mes 6 : 1.21 mm , Precipitacion Acumulada: 16.59 mm
Precipitacion del Mes 7 : 13.35 mm , Precipitacion Acumulada: 29.94 mm
Precipitacion del Mes 8 : 27.81 mm , Precipitacion Acumulada: 57.75 mm
Precipitacion del Mes 9 : 16.18 mm , Precipitacion Acumulada: 73.92 mm
Precipitacion del Mes 10 : 7.98 mm , Precipitacion Acumulada: 81.91 mm
Precipitacion del Mes 11 : 3.58 mm , Precipitacion Acumulada: 85.48 mm
Precipitacion del Mes 12 : 13.29 mm , Precipitacion Acumulada: 98.77 mm


**Meses mas lluviosos:** agosto con 27.81 mm, septiembre con 16.18 mm y julio con 13.35 mm


In [20]:
#Loop para calcular la temperatura maxima y minima promedio de cada mes 
#Temperatura minima
for i in range(12): #contador que va de 1 a 12 para los meses 
    TminPromMensual = df[df.MES==i+1]['TMIN'].sum()/df[df.MES ==i+1]['TMIN'].count()#Sumatoria las temperaturas registradas
                                                                                    #por mes y divida entre el numero toral de datos
    print("Temperatura minima del Mes", i+1,":", np.round(TminPromMensual, decimals=2), "ºC") #Impresion del resultado redondeado a 2 decimales
  
print(" ")
#Temperatura maxima
for i in range(12):
    TmaxPromMensual = df[df.MES==i+1]['TMAX'].sum()/df[df.MES==i+1]['TMAX'].count()
    print("Temperatura maxima del Mes", i+1,":", np.round(TmaxPromMensual, decimals=2), "ºC")

Temperatura minima del Mes 1 : 6.59 ºC
Temperatura minima del Mes 2 : 7.17 ºC
Temperatura minima del Mes 3 : 8.59 ºC
Temperatura minima del Mes 4 : 10.7 ºC
Temperatura minima del Mes 5 : 13.23 ºC
Temperatura minima del Mes 6 : 18.23 ºC
Temperatura minima del Mes 7 : 23.73 ºC
Temperatura minima del Mes 8 : 23.79 ºC
Temperatura minima del Mes 9 : 21.98 ºC
Temperatura minima del Mes 10 : 15.9 ºC
Temperatura minima del Mes 11 : 9.86 ºC
Temperatura minima del Mes 12 : 6.89 ºC
 
Temperatura maxima del Mes 1 : 19.7 ºC
Temperatura maxima del Mes 2 : 21.46 ºC
Temperatura maxima del Mes 3 : 22.93 ºC
Temperatura maxima del Mes 4 : 25.52 ºC
Temperatura maxima del Mes 5 : 28.09 ºC
Temperatura maxima del Mes 6 : 31.99 ºC
Temperatura maxima del Mes 7 : 33.01 ºC
Temperatura maxima del Mes 8 : 33.83 ºC
Temperatura maxima del Mes 9 : 33.13 ºC
Temperatura maxima del Mes 10 : 29.52 ºC
Temperatura maxima del Mes 11 : 24.32 ºC
Temperatura maxima del Mes 12 : 20.9 ºC


**El mes más caliente**: El mes de agosto con 33.83 ºC

**El mes más frio:** El mes de Enero con 6.59 ºC

In [21]:
#Loop para calcular la precipitacion anual y la cantidad de datos que se tienen por año
for i in range(1973,2011): # Contador que va desde el inicio de mis datos hasta el final en años 
    PrecipAnual = df['PRECIP'][df.AÑO==[i+1]].sum() #Sumatoria de la precipitacion por años
    NumDatos= df['PRECIP'][df.AÑO==[i+1]].count() #Calculo de la cantidad de datos que se tienen por año
    print("En el Año", i+1,"llovieron:", PrecipAnual, "mm,", "Este año se tienen", NumDatos, "datos,", np.round(NumDatos/365.0, decimals=2), "%") #Impresion de los resultados calculados 
    

En el Año 1974 llovieron: 89.8 mm, Este año se tienen 184 datos, 0.5 %
En el Año 1975 llovieron: 55.9 mm, Este año se tienen 365 datos, 1.0 %
En el Año 1976 llovieron: 106.9 mm, Este año se tienen 366 datos, 1.0 %
En el Año 1977 llovieron: 104.7 mm, Este año se tienen 365 datos, 1.0 %
En el Año 1978 llovieron: 54.0 mm, Este año se tienen 243 datos, 0.67 %
En el Año 1979 llovieron: 43.0 mm, Este año se tienen 122 datos, 0.33 %
En el Año 1980 llovieron: 69.5 mm, Este año se tienen 366 datos, 1.0 %
En el Año 1981 llovieron: 91.0 mm, Este año se tienen 365 datos, 1.0 %
En el Año 1982 llovieron: 153.5 mm, Este año se tienen 365 datos, 1.0 %
En el Año 1983 llovieron: 214.2 mm, Este año se tienen 334 datos, 0.92 %
En el Año 1984 llovieron: 263.9 mm, Este año se tienen 366 datos, 1.0 %
En el Año 1985 llovieron: 53.0 mm, Este año se tienen 365 datos, 1.0 %
En el Año 1986 llovieron: 9.0 mm, Este año se tienen 207 datos, 0.57 %
En el Año 1987 llovieron: 0.0 mm, Este año se tienen 0 datos, 0.0 %
E

**El año mas lluvioso:** 1992 con 279 mm de lluvia, se tienen el 84% de los datos de este año.

**El año mes seco:** 2010 se registraron 0 mm de lluvia pero solo se tienen el 32% de los datos de este año, 1986 con 9 mm de lluvia, aunque solo se tiene el 57% de los datos de este año
